website/docs/r/oauth_client.html.markdown

```diff
@@ -51,8 +72,17 @@ The following arguments are supported:
   `https://api.github.com` or `https://ghe.example.com/api/v3`).
 * `http_url` - (Required) The homepage of your VCS provider (e.g.
   `https://github.com` or `https://ghe.example.com`).
-* `oauth_token` - (Required) The token string you were given by your VCS provider, e.g. `ghp_xxxxxxxxxxxxxxx` for a GitHub personal access token. For more information on how to generate this token string for your VCS provider, see the [Create an OAuth Client](https://www.terraform.io/docs/cloud/api/oauth-clients.html#create-an-oauth-client) documentation.
+* `oauth_token` - The token string you were given by your VCS provider, e.g. `ghp_xxxxxxxxxxxxxxx` for a GitHub personal access token. For more information on how to generate this token string for your VCS provider, see the [Create an OAuth Client](https://www.terraform.io/docs/cloud/api/oauth-clients.html#create-an-oauth-client) documentation.
```

Comment:
In the docs we mention _when_ other fields are required, like "(Required for `ado_server`)" -- should we do the same here or is it never required now?

Reply:
Good catch! I didn't put a `Required for X` here because the default is to use an oauth_token. And by default, it mostly means GitHub and GitLab. I'd rather list the `Required for X` when its the exception for one provider, than to list it for multiple providers (or the default).

Also, this will be changing in the near term future as better terminology is used to match OAuth2 terminology. So i'll keep this for now